nifi-nar-bundles/nifi-marklogic-bundle/nifi-marklogic-processors/src/test/java/org/apache/nifi/marklogic/processor/QueryMarkLogicIT.java

```diff
@@ -96,10 +126,169 @@ public void testSimpleCollectionQuery() throws InitializationException {
         assertTrue(Arrays.equals(expectedByteArray, actualByteArray));
     }
 
+    @Test
+    public void testCombinedJSONQuery() throws InitializationException {
+        TestRunner runner = getNewTestRunner(QueryMarkLogic.class);
+        runner.setProperty(QueryMarkLogic.QUERY, "{\"search\" : {\n" +
+                "  \"ctsquery\": {\n" +
+                "    \"jsonPropertyValueQuery\":{\n" +
+                "      \"property\":[\"sample\"],\n" +
+                "      \"value\":[\"jsoncontent\"]\n" +
+                "      } \n" +
+                "  }\n" +
+                "} }");
+        runner.setProperty(QueryMarkLogic.QUERY_TYPE, QueryMarkLogic.QueryTypes.COMBINED_JSON.name());
+        runner.assertValid();
+        runner.run();
+        // mod 3 == 0 docs are JSON, but mod 5 == 0 docs are XML and take precedence in doc generation
+        int expectedSize = (int) (Math.ceil((numDocs - 1.0) / 3.0) - Math.ceil((numDocs - 1.0) / 15.0));
+        runner.assertTransferCount(QueryMarkLogic.SUCCESS, expectedSize);
+        runner.assertAllFlowFilesContainAttribute(QueryMarkLogic.SUCCESS,CoreAttributes.FILENAME.key());
+        List<MockFlowFile> flowFiles = runner.getFlowFilesForRelationship(QueryMarkLogic.SUCCESS);
+        assertEquals(flowFiles.size(), expectedSize);
+        byte[] actualByteArray = null;
+        for(MockFlowFile flowFile : flowFiles) {
+            if(flowFile.getAttribute(CoreAttributes.FILENAME.key()).endsWith("3.json")) {
```

Comment:
Not wild about the magic string here and magic number on line 156, but for a test it's okay.